vesper.go

```diff
@@ -13,23 +13,48 @@ var log logPrinter = noOpPrinter{}
 
 // Vesper is a middleware adapter for Lambda Functions
 type Vesper struct {
-	handler lambdaHandler
+	rawHandler    interface{}
+	middlewares   []Middleware
+	autoUnmarshal bool
 }
 
 // New creates a new Vesper instance given a Handler and set of Middleware
 // Middlewares are evaluated in the order they are provided
-func New(l interface{}, middlewares ...Middleware) *Vesper {
-	m := buildChain(newTypedToUntypedWrapper(l), middlewares...)
-	f := newMiddlewareWrapper(l, m)
+func New(handler interface{}, middlewares ...Middleware) *Vesper {
+	v := Vesper{
+		rawHandler:    handler,
+		middlewares:   middlewares,
+		autoUnmarshal: true,
+	}
+	return &v
+}
+
+func (v *Vesper) DisableAutoUnmarshal() *Vesper {
+	v.autoUnmarshal = false
+	return v
+}
+
+func (v *Vesper) AddMiddlewares(middlewares ...Middleware) *Vesper {
+	v.middlewares = append(v.middlewares, middlewares...)
+	return v
+}
+
+func (v *Vesper) AddMiddleware(middleware Middleware) *Vesper {
```

Comment:
Just wondering, would `Use` be a more idiomatic/consistent name here (e.g. https://github.com/gorilla/mux#middleware)? 

I think you can also just collapse the two `AddMiddleware*` functions into the single variadic form - unless I'm missing a use case here?

Reply:
That's a good idea 😄. It's also consistent with middyjs.

Will make the change when I get the chance!

Reply:
Awesome